pipelines/verify_execution.py

```
@@ -0,0 +1,326 @@
+import json
+import sys
+from pathlib import Path
+
+# Reuse existing validators as ground truth.
+try:
+    from check_records import check_record, ALLOWED_PROVENANCE
+    from validate_run import event_time
+except ImportError:
+    ALLOWED_PROVENANCE = frozenset({"designed", "simulated", "hil"})
+
+    def event_time(event):  # type: ignore
```

Ruff (0.16.1)

[warning] Unused function argument: `event`

(ARG001)